Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -507,17 +502,24 @@ void CookieStore::set(CookieInit&& options, Ref<DeferredPromise>&& promise)
     }
 
     cookie.path = WTFMove(options.path);
-    if (!cookie.path.isNull()) {
-        if (!cookie.path.startsWith('/')) {
-            promise->reject(Exception { ExceptionCode::TypeError, "The path must begin with a '/'"_s });
-            return;
-        }
+    ASSERT(!cookie.path.isNull());
+    if (cookie.path.isEmpty())
+        cookie.path = CookieUtil::defaultPathForURL(url);
 
-        // FIXME: <rdar://85515842> Obtain the encoded length without allocating and encoding.
-        if (cookie.path.utf8().length() > maximumAttributeValueSize) {
-            promise->reject(Exception { ExceptionCode::TypeError, makeString("The size of the path must not be greater than "_s, maximumAttributeValueSize, " bytes"_s) });
-            return;
-        }
+    if (!cookie.path.startsWith('/')) {
+        promise->reject(Exception { ExceptionCode::TypeError, "The path must begin with a '/'"_s });
+        return;
+    }
+
+    if (cookie.path != "/"_s && cookie.name.startsWithIgnoringASCIICase("__Host-"_s)) {
+        promise->reject(Exception { ExceptionCode::TypeError, "If the cookie name begins with \"__Host-\", the eventual path must be \"/\"."_s });
```

Comment:
Nit: What's the point of "eventual" here? Perhaps it's most clear to simply say `"... the path must be "/".`

Reply:
Actually let's keep what the code had before: `If the cookie name begins with "__Host-", the path must either not be specified or be "/".` because this matches Chrome.